src/Maker/Security/MakeJsonLogin.php

```diff
@@ -0,0 +1,106 @@
+{
+    public static function getCommandName(): string
+    {
+        return 'make:security:json-login';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command->setHelp(file_get_contents(\dirname(__DIR__, 2).'/Resources/help/security/MakeJsonLogin.txt'));
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Generate the code needed for the json_login authenticator';
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator): void
+    {
+        $userClassDetails = new ClassNameDetails($this->userClass, '');
+
+        $useStatements = new UseStatementGenerator([
+            $userClassDetails->getFullName(),
+            \Symfony\Bundle\FrameworkBundle\Controller\AbstractController::class,
+            \Symfony\Component\HttpFoundation\JsonResponse::class,
+            \Symfony\Component\HttpFoundation\Response::class,
+            \Symfony\Component\Routing\Annotation\Route::class,
+            \Symfony\Component\Security\Http\Attribute\CurrentUser::class,
+        ]);
+
+        $controllerNameDetails = $generator->createClassNameDetails($this->securityControllerName, 'Controller\\', 'Controller');
+
+        $controllerPath = $this->fileManager->getRelativePathForFutureClass($controllerNameDetails->getFullName());
+
+        $controllerExists = $this->fileManager->fileExists($controllerPath);
+
+        if (!$controllerExists) {
+            $generator->generateController(
+                $controllerNameDetails->getFullName(),
+                'EmptyController.tpl.php',
+                [
+                    'use_statements' => $useStatements,
+                    'controller_name' => $controllerNameDetails->getShortName(),
+                ]
+            );
+        }
+
+        $controllerSource = $controllerExists ? file_get_contents($controllerPath) : $generator->getFileContentsForPendingOperation($controllerPath);
+
+        $manipulator = new ClassSourceManipulator($controllerSource);
+
+        $this->securityControllerBuilder->addJsonLoginMethod($manipulator, $userClassDetails);
+
+        $securityData = $this->securityConfigUpdater->updateForJsonLogin($this->ysm->getContents(), $this->firewallToUpdate, 'app_api_login');
+
+        if ($this->willLogout) {
+            $this->securityControllerBuilder->addLogoutMethod($manipulator);
+
+            $securityData = $this->securityConfigUpdater->updateForLogout($securityData, $this->firewallToUpdate);
+        }
+
+        $generator->dumpFile(self::SECURITY_CONFIG_PATH, $securityData);
+        $generator->dumpFile($controllerPath, $manipulator->getSourceCode());
+
+        $generator->writeChanges();
+
+        $this->writeSuccessMessage($io);
+
+        $io->text([
+            'Next: Make a <info>POST</info> request to <info>/api/login</info> with a <info>username</info> and <info>password</info> to login.',
```

Comment:
The user also needs to set `Content-Type: application/json` iirc. This is sometimes a "gotcha", so it would be nice to mention it here.